src/ast/passes/codegen_llvm.cpp

```diff
@@ -3329,17 +3329,15 @@ void CodegenLLVM::createMapDefinition(const std::string &name,
 void CodegenLLVM::generate_maps(const RequiredResources &resources)
 {
   // User-defined maps
-  for (auto &map_val : resources.map_vals) {
-    const std::string &name = map_val.first;
+  for (auto &map_info : resources.maps_info) {
```

Comment:
nit: could destructure here. eg:

```
for (const auto &[name, info] : resources.maps_info) {
    ...
}
```

Reply:
Love this syntax! Would be nice to start using it more across the codebase.

Reply:
I suspect clang-tidy can probably catch this. I'll make a note in the tracking issue and take a look when I get a chance